osu.Game/Graphics/UserInterface/OsuPasswordTextBox.cs

```diff
@@ -15,6 +19,35 @@ public class OsuPasswordTextBox : OsuTextBox
 
         public override bool AllowClipboardExport => false;
 
+        private readonly CapsWarning warning;
+
+        private GameHost host;
+
+        public OsuPasswordTextBox()
+        {
+            Add(warning = new CapsWarning
+            {
+                Size = new Vector2(20),
+                Origin = Anchor.CentreRight,
+                Anchor = Anchor.CentreRight,
+                Margin = new MarginPadding { Right = 10 },
+                Alpha = 0,
+            });
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(GameHost host)
+        {
+            this.host = host;
+        }
+
+        protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)

```

Comment:
Shouldn't the warning be updated regardless of whether caps is pressed while this textbox is focused?

Reply:
Pretty sure it should start displaying when the `TextBox` receives focus, and hide after losing focus.

Reply:
_done_